### PR TITLE
Add Z heigth as footer item

### PR DESCRIPTION
**Pull request by MistrKernnunos**
BFW-2015

### CODE REVIEW

File: src/common/marlin_vars.h [hunk 1]
```diff
@@ -37,25 +37,31 @@ enum {
     MARLIN_VAR_TIMTOEND = 0x1d,            // R:  uint32, oProgressData.oTime2End.mGetValue() or -1 if not valid
     MARLIN_VAR_FAN0_RPM = 0x1e,            // R:  uint16, fanctl0.getActualRPM()
     MARLIN_VAR_FAN1_RPM = 0x1f,            // R:  uint16, fanctl1.getActualRPM()
-    MARLIN_VAR_FAN_CHECK_ENABLED = 0x20,   //RW: uintt8, fan_check
-    MARLIN_VAR_FS_AUTOLOAD_ENABLED = 0x21, //RW: uint8_t fs_autoload
-    MARLIN_VAR_MAX = MARLIN_VAR_FS_AUTOLOAD_ENABLED
+    MARLIN_VAR_FAN_CHECK_ENABLED = 0x20,   // RW: uintt8, fan_check
+    MARLIN_VAR_FS_AUTOLOAD_ENABLED = 0x21, // RW: uint8_t fs_autoload
+    MARLIN_VAR_CURR_POS_X = 0x22,          // R: float current_position
```
Comment (RadekVana) [inline src/common/marlin_vars.h:42]: R or RW?
why is X different from Y Z E?

File: src/common/marlin_vars.h [hunk 2]
```diff
@@ -37,25 +37,31 @@ enum {
...
+    MARLIN_VAR_CURR_POS_E = 0x25,          // RW: ==||==
+    MARLIN_VAR_MAX = MARLIN_VAR_CURR_POS_E
 };
 
 // variable masks
 #define MARLIN_VAR_MSK(v_id) (((uint64_t)1) << (uint8_t)(v_id))
 
-//maximum number of masks is 64
-//maximum mask index is 63
+// maximum number of masks is 64
```
Comment (RadekVana) [inline src/common/marlin_vars.h:52]: don't add spaces to comments

File: src/common/marlin_vars.h [hunk 3]
```diff
@@ -150,6 +156,7 @@ typedef struct _marlin_vars_t {
     uint8_t media_inserted;      // media_is_inserted()
     uint8_t fan_check_enabled;   // fan_check [on/off]
     uint8_t fs_autoload_enabled; // fs_autoload [on/off]
+    float curr_pos[4];           // current position XYZE according to G-code [mm]
```
Comment (RadekVana) [inline src/common/marlin_vars.h:159]: 4 floats does not have size 1B .. move it up
order is important - affects size

File: src/gui/footer/footer_item_z_heigth.cpp [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+string_view_utf8 FooterItemZHeigth::static_makeView(float value) {
+    static std::array<char, 7> buff;
+    float value_clamped = std::clamp((float)value, 0.f, 100.f);
+
+    int printed_chars = snprintf(buff.data(), buff.size(), "%.2f", (double)value_clamped);
+
+    if (printed_chars < 1) {
+        buff[0] = '\0';
+    } else {
+        while ((--printed_chars) > 2 && buff[printed_chars] == '0' && buff[printed_chars - 1] != '.') {
```
Comment (RadekVana) [inline src/gui/footer/footer_item_z_heigth.cpp:16]: I'm not sure about operator `>` `&&` and `==` priority
please consider adding brackets

File: src/gui/footer/footer_item_z_heigth.cpp [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            buff[printed_chars] = '\0';
+        }
+    }
+    return string_view_utf8::MakeRAM((const uint8_t *)buff.data());
```
Comment (RadekVana) [inline src/gui/footer/footer_item_z_heigth.cpp:20]: consider comment that buffer must be static for

**Review by RadekVana — CHANGES_REQUESTED** [review]

File: src/common/marlin_vars.h (hunk 1 above)
Comment (MistrKernnunos) [reply src/common/marlin_vars.h:42]: 👍

File: src/common/marlin_vars.h (hunk 2 above)
Comment (MistrKernnunos) [reply src/common/marlin_vars.h:52]: 👍

File: src/common/marlin_vars.h (hunk 3 above)
Comment (MistrKernnunos) [reply src/common/marlin_vars.h:159]: 👍

File: src/gui/footer/footer_item_z_heigth.cpp (hunk 4 above)
Comment (MistrKernnunos) [reply src/gui/footer/footer_item_z_heigth.cpp:16]: 👍

File: src/gui/footer/footer_item_z_heigth.cpp (hunk 5 above)
Comment (MistrKernnunos) [reply src/gui/footer/footer_item_z_heigth.cpp:20]: 👍

**Review by RadekVana — APPROVED** [review]

**Pull request merged**